### PR TITLE
Migrate hdf5 version

### PR DESCRIPTION
**Pull request by hmaarrfk**
Blocked on: https://github.com/conda-forge/netcdf-cxx4-feedstock/pull/35

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
A new version will be needed and there's no indication of when that might happen:
https://github.com/Unidata/netcdf-cxx4/issues/106

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

**Comment by hmaarrfk** [comment]
With enough patching, anything is possible!

**Comment by hmaarrfk** [comment]
@xylar for some reason osx find HDF5 but then fails at defining the variables.

Maybe we can help CMake find HDF5 somehow?

**Comment by xylar** [comment]
@hmaarrfk, sorry, I really don't have much expertise at all in Cmake.  I'm not sure I'm going to be much help here.  Seems pretty frustrating!

**Comment by hmaarrfk** [comment]
The "not enough" argument error is a red herring. The real error is that `FindHDF5` does not set the variables correctly

**Comment by xylar** [comment]
@hmaarrfk, you got it! At least it looks that way!

**Comment by kmuehlbauer** [comment]
@hmaarrfk Great sleuthing!

**Comment by hmaarrfk** [comment]
Merging this will basically close out the HDF5 migration. It doesn't seem like there is much else that needs to be moved over.

**Review by xylar — APPROVED** [review]
Looks good to me!

**Comment by xylar** [comment]
@ocefpaf and @kmuehlbauer, let me know if you have any concerns.  I'll merge this unless I hear something soon (by tomorrow).

File: recipe/meta.yaml [hunk 1]
```diff
@@ -69,3 +86,4 @@ extra:
     - kmuehlbauer
     - ocefpaf
     - xylar
+    - hmaarrfk
```
Comment (ocefpaf) [inline recipe/meta.yaml:89]: :+1:

**Comment by ocefpaf** [comment]
> @ocefpaf and @kmuehlbauer, let me know if you have any concerns. I'll merge this unless I hear something soon (by tomorrow).

LGTM! Merge away!!

Thanks @hmaarrfk!

**Pull request merged**